miasm2/os_dep/common.py

```diff
@@ -72,10 +73,14 @@ def vm_alloc(self, vm, size, perm=PAGE_READ | PAGE_WRITE):
         @perm: permission flags (PAGE_READ, PAGE_WRITE, PAGE_EXEC or any `|`
             combination of them); default is PAGE_READ|PAGE_WRITE
         """
-        addr = self.next_addr(size)
-        vm.add_memory_page(addr, perm, "\x00" * size,
+        addr = self.next_addr(size+8)
+        vm.add_memory_page(addr, perm, "\x00" * (size+8),
```

Comment:
Mixing a meta-info such as the page size with the data is probably not a good idea.
This is more a "structural" information, such as memory access rights, and it is already save in the implementation of the memory pages.
For now, there is no direct API to get it back, but you can obtain it by using `vm.get_all_memory()[start_addr]["size"]`.
An API like the one with mem access is maybe missing.

Reply:
The API you give is linked to memory pages, and not the heap that is used here no? This would mean that the size returned by ``vm.get_all_memory()[start_addr]["size"]`` would be the size of the page where the heap resides, but not the size of the allocated heap chunk.